src/vs/code/node/cliProcessMain.ts

```diff
@@ -361,8 +361,20 @@ export async function main(argv: ParsedArgs): Promise<void> {
 
 		try {
 			await main.run(argv);
+
 			// Flush the remaining data in AI adapter.
-			await combinedAppender(...appenders).flush();
+			// If it does not complete in 1 second, exit the process.
+			const flushTelemetry = (promise: Promise<any>, ms: number): Promise<any> => {
+				const timeout = new Promise((resolve) => {
```

Comment:
Check this out: https://github.com/microsoft/vscode/blob/613bcf26c1cb4c3d4735a34c660bc5fea7e0bbae/src/vs/base/common/async.ts#L59-L59